src/sdam/topology.ts

```diff
@@ -1053,3 +1054,10 @@ export class ServerCapabilities {
     return this.maxWireVersion >= 5;
   }
 }
+
+/** @internal */
+export function getTopology(provider: MongoClient | Db | Collection): Topology | undefined {
```

Comment:
Another way to implement this is to use ducktyping. It would remove the need for having to import these types into the file directly, which will prevent cycles in module loading:
```
export function getTopology(provider: MongoClient | Db | Collection): Topology {
  if (`topology` in provider) return provider.topology;
  if ('s' in provider && 'client' in provider.s && 'topology' in provider.s.client) return provider.s.client.topology;
  ...
```